File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/routes/home/HomePageDTO.java
package com.chaincuerealestate.chaincuerealestate.routes.home;

public record HomePageDTO(
        Country[] countries,
        House[] recentlyAddedHouses
) {
    record Country(
            String name
    ) {
    }

    record House(
            String id,
            String title,
            String location,
            int numberRooms,
            int beds,
            String dollarPrice,
            String cryptoPrice,
            String src
    ) {
    }
}

File: kotlin/chaincue-real-estate-reactive/src/main/kotlin/com/chaincuerealestate/chaincuerealestatereactive/routes/home/HomePageDTO.kt
package com.example.chaincuerealestate.routes.home

data class HomePageDTO(
    val countries: Array<Country>,
    val recentlyAddedHouses: Array<House>
) {
    data class Country(
        val name: String
    )

    data class House(
        val id: String,
        val title: String,
        val location: String,
        val numberRooms: Int,
        val beds: Int,
        val dollarPrice: String,
        val cryptoPrice: String,
        val src: String
    )
}
